src/libraries/Microsoft.Extensions.Http/src/DefaultHttpClientFactory.cs

```diff
@@ -313,6 +313,75 @@ internal void CleanupTimer_Tick()
             }
         }
 
+        public void Dispose()
+        {
+            // Stop the cleanup timer first to prevent any more items from being processed
+            lock (_cleanupTimerLock)
+            {
+                if (_cleanupTimer != null)
+                {
+                    _cleanupTimer.Dispose();
+                    _cleanupTimer = null;
+                }
+            }
+
+            // Stop all active handler timers to prevent more entries being added to _expiredHandlers
+            foreach (var pair in _activeHandlers)
+            {
+                if (pair.Value.IsValueCreated)
+                {
+                    pair.Value.Value.StopTimer();
+                }
+            }
+
+            // Process all expired handlers immediately
+            if (!Monitor.TryEnter(_cleanupActiveLock))
+            {
+                return; // Another thread is already cleaning up
+            }
+
+            try
+            {
+                while (_expiredHandlers.TryDequeue(out ExpiredHandlerTrackingEntry? entry))
+                {
+                    try
+                    {
+                        // Force dispose regardless of CanDispose
+                        entry.InnerHandler.Dispose();
+                        entry.Scope?.Dispose();
+                    }
+                    catch (Exception ex)
+                    {
+                        Log.CleanupItemFailed(_logger, entry.Name, ex);
+                    }
+                }
+
+                // Dispose all active handlers
+                foreach (var pair in _activeHandlers)
+                {
+                    if (pair.Value.IsValueCreated)
+                    {
+                        var entry = pair.Value.Value;
+                        try
+                        {
+                            entry.Handler.InnerHandler!.Dispose();
```

Comment:
Does this change mean that `HttpClient` instances created by `DefaultHttpClientFactory` are now effectively scoped to the lifetime of the factory instance?

If so, was that always the intent, and the current behavior incorret? 
Or is this a breaking change that will require additional communication and documentation?